File: apps/python-sync/app/models/sync_receipt_log.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, Text
from sqlalchemy.dialects.mysql import JSON
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class SyncReceiptLog(Base):
    __tablename__ = "sync_receipt_logs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    sync_task_id: Mapped[int] = mapped_column(Integer, nullable=False)
    request_id: Mapped[str] = mapped_column(String(64), nullable=False)
    phase: Mapped[str] = mapped_column(String(32), nullable=False)
    http_status: Mapped[int | None] = mapped_column(Integer, nullable=True)
    platform_code: Mapped[str] = mapped_column(String(64), nullable=False)
    endpoint: Mapped[str | None] = mapped_column(String(255), nullable=True)
    success: Mapped[bool | None] = mapped_column(Boolean, nullable=True)
    accepted: Mapped[bool | None] = mapped_column(Boolean, nullable=True)
    final_result: Mapped[bool | None] = mapped_column(Boolean, nullable=True)
    external_id: Mapped[str | None] = mapped_column(String(128), nullable=True)
    code: Mapped[str | None] = mapped_column(String(64), nullable=True)
    message: Mapped[str | None] = mapped_column(Text, nullable=True)
    request_payload: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    response_payload: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    created_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)

File: apps/php-api/app/Models/SyncReceiptLog.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class SyncReceiptLog extends Model
{
    protected $table = 'sync_receipt_logs';

    public $timestamps = false;

    protected $fillable = [
        'sync_task_id',
        'request_id',
        'phase',
        'http_status',
        'platform_code',
        'endpoint',
        'success',
        'accepted',
        'final_result',
        'external_id',
        'code',
        'message',
        'request_payload',
        'response_payload',
        'created_at',
    ];

    protected function casts(): array
    {
        return [
            'success' => 'boolean',
            'accepted' => 'boolean',
            'final_result' => 'boolean',
            'request_payload' => 'array',
            'response_payload' => 'array',
            'created_at' => 'datetime',
        ];
    }
}
